File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.go
func lengthOfLongestSubstring(s string) int {
  ans, l := -1, 0
  chars := map[uint8]bool{}
  for r := range s {
    for chars[s[r]] {
      delete(chars, s[l])
      l += 1
    }
    chars[s[r]] = true
    ans = max(ans, r - l)
  }
  return ans + 1
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.ts
function lengthOfLongestSubstring(s: string): number {
  let ans = -1;
  for (let set = new Set<string>(), l = 0, r = 0; r < s.length; r++) {
    while (set.has(s[r])) set.delete(s[l++])
    set.add(s[r])
    if (ans < r - l) {
      ans = r - l;
    }
  }
  return ans + 1;
};